modular_skyrat/modules/modsuit_pai/code/mod_pai.dm

```diff
@@ -0,0 +1,137 @@
+/**
+ * Simple proc to insert the pAI into the MODsuit.
+ *
+ * user - The person trying to put the pAI into the MODsuit.
+ * card - The pAI card we're slotting in the MODsuit.
+ */
+
+/obj/item/mod/control/proc/insert_pai(mob/user, obj/item/paicard/card)
+	if(mod_pai)
+		balloon_alert(user, "pAI already installed!")
+		return
+	if(!card.pai || !card.pai.mind)
+		balloon_alert(user, "pAI unresponsive!")
+		return
+	balloon_alert(user, "transferring to suit...")
+	if(!do_after(user, 5 SECONDS, target = src))
+		balloon_alert(user, "interrupted!")
+		return FALSE
+	if(!user.transferItemToLoc(card, src))
+		return
+
+	mod_pai = card.pai
+	balloon_alert(user, "pAI transferred to suit")
+	balloon_alert(mod_pai, "transferred to a suit")
+	mod_pai.can_transmit = TRUE
+	mod_pai.can_receive = TRUE
+	mod_pai.canholo = FALSE
+	mod_pai.remote_control = src
+	for(var/datum/action/action as anything in actions)
+		action.Grant(mod_pai)
+	return TRUE
+
+/**
+ * Simple proc to remove the pAI from the MODsuit.
+ *
+ * user - The person trying to take out the pAI from the MODsuit.
+ * forced - Whether or not we skip the checks and just eject the pAI. Defaults to FALSE.
+ */
+/obj/item/mod/control/proc/remove_pai(mob/user, forced = FALSE)
+	if(!mod_pai)
+		if(user)
+			balloon_alert(user, "no pAI to remove!")
+		return
+	if(!forced)
+		if(!open)
+			balloon_alert(user, "open the suit panel!")
+			return FALSE
+		if(!do_after(user, 5 SECONDS, target = src))
+			balloon_alert(user, "interrupted!")
+			return FALSE
+
+	var/turf/drop_off = get_turf(src)
+	if(drop_off) // In case there's no drop_off, the pAI will simply get deleted.
+		mod_pai.card.forceMove(drop_off)
+
+	for(var/datum/action/action as anything in actions)
```

Comment:
What if the actions in the suit change? The action button will remain?

Reply:
The actions won't change unless it's pinned actions, which require the pAI user to intentionally do that, and I'd assume that's already handled by the action itself, and if it's not, we can test that and I can push a fix upstream, because they'll likely have the same issue with MODsuit AIs.

Reply:
That was fixed upstream.